File: php/features.php
<?php
declare(strict_types=1);

// BrewpageHosting SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BrewpageHostingFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BrewpageHostingBaseFeature();
            case "test":
                return new BrewpageHostingTestFeature();
            default:
                return new BrewpageHostingBaseFeature();
        }
    }
}

File: py/features.py
# BrewpageHosting SDK feature factory

from feature.base_feature import BrewpageHostingBaseFeature
from feature.test_feature import BrewpageHostingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BrewpageHostingBaseFeature(),
        "test": lambda: BrewpageHostingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
